### PR TITLE
Re-enable cinema mode

### PR DESCRIPTION
**Pull request by MrTimscampi**
**Changes**

I messed up #835, so here is the original contents of PR with my correction.

**Issues**

### CODE REVIEW

**Comment by dmitrylyzo** [comment]
Good news: it works with patch below
Bad news: `IsUnPlayed` is "never played before", not unwatched/unmarked

<details>
<summary>Quick fix (just to make it work)</summary>

```diff
diff --git a/src/components/playback/playbackmanager.js b/src/components/playback/playbackmanager.js
index b03a59380..45e1c2d76 100644
--- a/src/components/playback/playbackmanager.js
+++ b/src/components/playback/playbackmanager.js
@@ -334,13 +334,22 @@ define(['events', 'datetime', 'appSettings', 'itemHelper', 'pluginManager', 'pla
         return apiClient.getItems(apiClient.getCurrentUserId(), fetchOptions).then(function (unwatchedMovies) {
             var promises = [];
 
-            for (let unwatchedMovie in unwatchedMovies) {
+            for (let unwatchedMovie of unwatchedMovies.Items) {
                 const remoteTrailers = getRemoteTrailers(unwatchedMovie);
 
                 promises.push(getAllTrailers(unwatchedMovie, remoteTrailers));
             }
 
             return Promise.all(promises);
+        }).then(function (trailersOfUnwathedMovies) {
+            const introItems = [];
+
+            // Choose random trailer for each movie
+            for (let trailers of trailersOfUnwathedMovies) {
+                introItems.push(trailers[Math.floor(Math.random() * trailers.length)]);
+            }
+
+            return Promise.resolve({Items: introItems});
         }).catch(function (err) {
             console.error(`failed to get trailers: ${err}`);
 
diff --git a/src/scripts/site.js b/src/scripts/site.js
index ddb152eb4..6aa4213bc 100644
--- a/src/scripts/site.js
+++ b/src/scripts/site.js
@@ -845,7 +845,7 @@ var AppInfo = {};
         define('sortMenu', [componentsPath + '/sortmenu/sortmenu'], returnFirstDependency);
         define('sanitizefilename', [componentsPath + '/sanitizeFilename'], returnFirstDependency);
         define('toast', [componentsPath + '/toast/toast'], returnFirstDependency);
-        define('scrollHelper', [componentsPath + '/scrollHelper'], returnFirstDependency);
+        define('scrollHelper', [scriptsPath + '/scrollHelper'], returnFirstDependency);
         define('touchHelper', [scriptsPath + '/touchHelper'], returnFirstDependency);
         define('imageUploader', [componentsPath + '/imageUploader/imageUploader'], returnFirstDependency);
         define('htmlMediaHelper', [componentsPath + '/htmlMediaHelper'], returnFirstDependency);
```
</details>

**Comment by MrTimscampi** [comment]
@jellyfin/web We should really get this merged, the original one is from late February.

File: src/scripts/settings/userSettings.js [hunk 1]
```diff
@@ -89,9 +89,27 @@ import events from 'events';
         }
 
         val = this.get('enableCinemaMode', false);
+        if (val == null) return false;
         return val !== 'false';
```
Comment (dmitrylyzo) [inline src/scripts/settings/userSettings.js:93]: Not tested
```suggestion
        return val === 'true';
```

File: src/components/playback/playbackmanager.js [hunk 2]
```diff
@@ -279,17 +279,79 @@ define(['events', 'datetime', 'appSettings', 'itemHelper', 'pluginManager', 'pla
...
+            } else {
+                return Promise.resolve(remoteTrailers);
+            }
+        }
 
-            return result;
+        return apiClient.getItems(apiClient.getCurrentUserId(), fetchOptions).then(function (unwatchedMovies) {
+            var promises = [];
+
+            for (let unwatchedMovie in unwatchedMovies.Items) {
```
Comment (dmitrylyzo) [inline src/components/playback/playbackmanager.js:337]:
```suggestion
            for (let unwatchedMovie of unwatchedMovies.Items) {
```
_My patch has this_

**Comment by dmitrylyzo** [comment]
> The problem is that I see trailers of single movie.

`HasTrailer: true,` for some reason ignores remote trailers.
https://github.com/jellyfin/jellyfin/blob/d38adb95a727203b0d0dcee344f03b374b5d2b8f/MediaBrowser.Controller/Entities/UserViewBuilder.cs#L744-L750
_btw, we could use `hasTrailers.GetTrailerCount` - will save some nanoseconds :p_
https://github.com/jellyfin/jellyfin/blob/d38adb95a727203b0d0dcee344f03b374b5d2b8f/MediaBrowser.Controller/Entities/IHasTrailers.cs#L39-L40
This requires backend debugging.

> Hmm, after youtube trailer, videoosd doesn't want to be shown. There seems to be something wrong with switching players.

d7c2fd7337d35f6cf87d54b59769c663d3934f50
Everything works if we don't change players.
As an option, check for non-videoosd page.

File: src/components/playback/playbackmanager.js [hunk 3]
```diff
@@ -279,17 +279,79 @@ define(['events', 'datetime', 'appSettings', 'itemHelper', 'pluginManager', 'pla
...
 
-        return apiClient.getIntros(firstItem.Id).then(function (result) {
+        var fetchOptions = {
+            Fields: 'RemoteTrailers,LocalTrailerCount,ServerID',
+            Filters: unwatchedOnly ? 'IsUnPlayed' : '',
+            HasTrailer: true,
+            IncludeItemTypes: 'Movie',
+            Limit: trailerCount,
+            Recursive: true,
+            SortBy: 'Random'
```
Comment (JustAMan) [inline src/components/playback/playbackmanager.js:299]: I don't think we should be shuffling stuff server-side, we better do it client-side for server performance

File: src/components/playback/playbackmanager.js [hunk 4]
```diff
@@ -279,17 +279,79 @@ define(['events', 'datetime', 'appSettings', 'itemHelper', 'pluginManager', 'pla
...
+        function getRemoteTrailers(item) {
+            var remoteTrailers = item.RemoteTrailers || [];
+
+            if (!remoteTrailers.length) {
+                return [];
+            }
+
+            return remoteTrailers.map(function (t) {
+                return {
+                    Name: t.Name || (item.Name + ' Trailer'),
```
Comment (JustAMan) [inline src/components/playback/playbackmanager.js:311]: this `' Trailer'` string should be i18ed

File: src/components/playback/playbackmanager.js [hunk 5]
```diff
@@ -279,17 +279,79 @@ define(['events', 'datetime', 'appSettings', 'itemHelper', 'pluginManager', 'pla
...
+                    Type: 'Trailer',
+                    ServerId: apiClient.serverId()
+                };
+            });
+        }
+
+        function getAllTrailers(item, remoteTrailers) {
+            if (item.LocalTrailerCount) {
+                return apiClient.getLocalTrailers(apiClient.getCurrentUserId(), item.Id).then(function (trailers) {
+                    trailers.push.apply(trailers, remoteTrailers);
```
Comment (JustAMan) [inline src/components/playback/playbackmanager.js:323]: any reason _not_ to call `trailers.push(remoteTrailers)`?

Comment (dmitrylyzo) [reply src/components/playback/playbackmanager.js:323]: We should push elements of `remoteTrailers`, not the array itself, i.e merge arrays

File: src/components/playback/playbackmanager.js (hunk 3 above)
Comment (dmitrylyzo) [reply src/components/playback/playbackmanager.js:299]: ref https://github.com/jellyfin/jellyfin-web/pull/835#discussion_r399773115

**Pull request closed without merging**